services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/invocationlogs.py

```diff
@@ -69,25 +71,44 @@ def add_invocationlog(self, aef_id, invocationlog):
             if result is not None:
                 return result
 
+            current_app.logger.debug("URL aef_id: {}".format(aef_id))
+            current_app.logger.debug("URL invocationlog.aef_id: {}".format(invocationlog.aef_id))
+            if aef_id != invocationlog.aef_id:
+                return bad_request_error(detail="invocationlog.aef_id is not in the same with URL aef_id", cause="Detected different values of aef_id parameter", invalid_params=[{"param": "invocationlog.aef_id", "reason": "Not the same with URL aef_id"}])
+
             for i in range(0, len(invocationlog.logs)):
                 result = self.__check_service_apis(invocationlog.logs[i].api_id, invocationlog.logs[i].api_name)
 
                 if result is not None:
                     return result
 
-            log_id = secrets.token_hex(15)
-            rec = dict()
-            rec['log_id'] = log_id
-            rec.update(invocationlog.to_dict())
-            mycol.insert_one(rec)
+            my_query = {'aef_id': aef_id, 'api_invoker_id': invocationlog.api_invoker_id}
+            existing_invocationlog = mycol.find_one(my_query)
+
+            if existing_invocationlog is None:
+                current_app.logger.debug("Test 0")
```

Comment:
Why are you updating log_id?

Reply:
I don't update log_id here. I create it. This is the case where no similar InvocationLog is found. So i create the log_id